lib/src/models/transaction_uploader.dart

```diff
@@ -131,18 +132,23 @@ class TransactionUploader {
     //  throw StateError('Unable to validate chunk: $_chunkIndex');
 
     // Catch network errors and turn them into objects with status -1 and an error message.
-    final res = await _api.post('chunk', body: json.encode(chunk));
-
+    Response? res;
+    try {
+      res = await _api.post('chunk', body: json.encode(chunk));
+    } catch (e) {
+      print("Error posting to /chunk endpoint: " + e.toString());
+    }
     _lastRequestTimeEnd = DateTime.now().millisecondsSinceEpoch;
-    lastResponseStatus = res.statusCode;
-
-    if (lastResponseStatus == 200) {
-      _chunkIndex++;
-    } else {
-      lastResponseError = getResponseError(res);
-      if (FATAL_CHUNK_UPLOAD_ERRORS.contains(lastResponseError)) {
-        throw StateError(
-            'Fatal error uploading chunk: $_chunkIndex: $lastResponseError');
+    if (res != null) {
+      lastResponseStatus = res.statusCode;
+      if (lastResponseStatus == 200) {
+        _chunkIndex++;
+      } else {
```

Comment:
I still think we should fix the `getResponseError` implementation. It needs to return a string and isn't ensuring that rsp.body actually has data.

Reply:
getResponseError will thow an error. res.body is non nullable. It won't reach that state since we handle post errors before it so there will always be a response.

Reply:
Actually looking at it now I suppose we're trusting the return type from the Dart HTTP library. If we're satisfied that that's working correctly, then we're ok.